horace_core/@d1d/cut.m

```diff
@@ -48,14 +48,21 @@
 
 % Perform operations
 % ------------------
+if iscell(w.data)&& ischar(w.data{1})
+    % Slightly hacky - load dnd object before passing it to sqw.cut so we don't
+    % attempt to load the file in as an SQW file. We reach this point if cut_dnd
+    % is called.
+    % After DnDBase class is implemented, cutting a DnD object need no longer
+    % go through sqw.cut and this file will be largely re-written
+    w.data = d1d(w.data{1});
```

Comment:
This function is a bit confusing. The `horace_function_parse_input` function actually outputs a struct: so `w` is a struct. The `.data` attribute is "the data" passed into the function. It's either a dnd or a file name (that's in a cell array for a reason that's not clear to me). 

I didn't want to touch this code too much since hopefully, was we have the `DnDBase` class, this file can be deleted.

Reply:
If the file is confusing, a comment to express the confusion and how you expect it to be resolved could be helpful.

Reply:
this will/should all be changing with the new DnD classes